Observer/CreateProductFaqObserver.php

```diff
@@ -0,0 +1,149 @@
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Mail\Template\TransportBuilder;
+use Magento\Framework\Translate\Inline\StateInterface;
+use Magento\Store\Model\StoreManagerInterface;
+use Psr\Log\LoggerInterface;
+
+class CreateProductFaqObserver implements ObserverInterface
+{
+    private const TEMPLATE_IDENTIFIER = 'product_faq_notification';
+
+    /**
+     * @var ProductRepositoryInterface
+     */
+    protected $productRepository;
+
+    /**
+     * @var CustomerRepositoryInterface
+     */
+    protected $customerRepository;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    protected $storeManager;
+
+    /**
+     * @var TransportBuilder
+     */
+    protected $transportBuilder;
+
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * @var ScopeConfigInterface
+     */
+    protected $config;
+
+    /**
+     * @var StateInterface
+     */
+    protected $inlineTranslation;
+
+    /**
+     * CreateProductFaqObserver constructor.
+     * @param ProductRepositoryInterface $productRepository
+     * @param CustomerRepositoryInterface $customerRepository
+     * @param StoreManagerInterface $storeManager
+     * @param TransportBuilder $transportBuilder
+     * @param LoggerInterface $logger
+     * @param ScopeConfigInterface $config
+     * @param StateInterface $inlineTranslation
+     */
+    public function __construct(
+        ProductRepositoryInterface  $productRepository,
+        CustomerRepositoryInterface $customerRepository,
+        StoreManagerInterface       $storeManager,
+        TransportBuilder            $transportBuilder,
+        LoggerInterface             $logger,
+        ScopeConfigInterface        $config,
+        StateInterface              $inlineTranslation
+    ) {
+        $this->productRepository = $productRepository;
+        $this->customerRepository = $customerRepository;
+        $this->storeManager = $storeManager;
+        $this->transportBuilder = $transportBuilder;
+        $this->logger = $logger;
+        $this->config = $config;
+        $this->inlineTranslation = $inlineTranslation;
+    }
+
+    /**
+     * @param Observer $observer
+     * @throws LocalizedException
+     */
+    public function execute(Observer $observer)
+    {
+        /**
+         * @var ProductFaq $productFaq
+         */
+        $productFaq = $observer->getData('productFaq');
+
+        try {
+            $customer = $this->customerRepository->getById($productFaq->getCustomerId());
+        } catch (\Exception $e) {
+            $this->logger->error($e->getMessage());
+
+            return;
+        }
+
+        try {
+            $product = $this->productRepository->getById($productFaq->getProductId());
+        } catch (\Exception $e) {
+            $this->logger->error($e->getMessage());
+
+            return;
+        }
```

Comment:
Može ti sve ići u jedan try/catch. 

Kada radiš logging errora uvije napiš neki unique string da kasnije znaš naći gdje se taj error dogodio. Vjeruj mi, nema gorega kada vidiš error u log-u, a ne znaš gdje se dogodio.

Možda ti je bolje $customer i $product objecte postali s mjesta gdje se event dispatcha ako ih već imaš u tom trenutku.

```suggestion
        try {
            $customer = $this->customerRepository->getById($productFaq->getCustomerId());
            $product = $this->productRepository->getById($productFaq->getProductId());
        } catch (\Exception $e) {
            $this->logger->error("Problem with Faq send email, ID: '{$productFaq->getId()}'", [
                    'message' => $e->getMessage()
                ]);
            return;
        }
```